master/buildbot/mq/wamp.py

```diff
@@ -0,0 +1,96 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from autobahn.wamp.exception import TransportLost
+from autobahn.wamp.types import PublishOptions
+from autobahn.wamp.types import SubscribeOptions
+
+from twisted.internet import defer
+
+from buildbot.mq import base
+from buildbot.util import service
+from buildbot.util import toJson
+from twisted.python import log
+
+import json
+
+
+class WampMQ(service.ReconfigurableServiceMixin, base.MQBase):
```

Comment:
Same, this is the original design of the mq configuration. I think this is indeed not a bad idea to change it to BuildbotService eventually, but this will break a number of unit tests, so this should be seperated.